qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -396,11 +494,25 @@ def _run_curve_fit(
             if new.success and res.redchi > new.redchi:
                 res = new
 
+        residuals_model = []
+        if res.success and self.options.get("plot_residuals", None):
+            for weights in residual_weights_list:
+                if isinstance(weights, np.ndarray):
+                    residuals_model.append(
+                        [res / np.abs(weight) for res, weight in zip(res.residual, weights)]
+                    )
+                else:
+                    residuals_model.append(res.residual)
+
+        if residuals_model:
```

Comment:
I changed the `residuals_model` default value do it could be `None` which I think make it clearer to the user that there was no residuals calculation.